backend/src/main/resources/db/migration/V1.10__pipeline_version_per_organism.sql

```diff
@@ -0,0 +1,116 @@
+-- Introduced in https://github.com/loculus-project/loculus/pull/3534
+-- Adds the new organism column to the current_processing_pipeline table,
+-- Adds the organism constraint to external_metadata_view and sequence_entries_view
+
+ALTER TABLE current_processing_pipeline
+ADD COLUMN organism text;
+
+-- Add current versions for all organisms
+WITH distinct_organisms AS (
+    SELECT DISTINCT organism
+    FROM sequence_entries
+),
+pipeline_versions AS (
+    SELECT version, started_using_at
+    FROM current_processing_pipeline
+)
+INSERT INTO current_processing_pipeline (version, started_using_at, organism)
+SELECT pv.version, pv.started_using_at, o.organism
+FROM pipeline_versions pv
+CROSS JOIN distinct_organisms o;
+
+-- delete old null rows
+DELETE FROM current_processing_pipeline
+WHERE organism IS NULL;
+
+-- Now, enforce the NOT NULL constraint
+ALTER TABLE current_processing_pipeline
+ALTER COLUMN organism SET NOT NULL;
+
+-- Update the primary key to include the organism
+ALTER TABLE current_processing_pipeline
+DROP CONSTRAINT current_processing_pipeline_pkey;
+ALTER TABLE current_processing_pipeline
+ADD CONSTRAINT current_processing_pipeline_pkey PRIMARY KEY (organism);
+
+drop view if exists external_metadata_view cascade;
+
+create view external_metadata_view as
+select
+    cpd.accession,
+    cpd.version,
+    all_external_metadata.updated_metadata_at,
+    -- Combines metadata from preprocessed data with any external metadata updates
+    -- If there's no external metadata, just use the preprocessed data's metadata
+    -- If there is external metadata, merge it with preprocessed data (external takes precedence)
+    case
+        when all_external_metadata.external_metadata is null then
+            jsonb_build_object('metadata', (cpd.processed_data->'metadata'))
+        else
+            jsonb_build_object(
+                'metadata',
+                (cpd.processed_data->'metadata') || all_external_metadata.external_metadata
+            )
+    end as joint_metadata
+from
+    (
+        -- Get only the preprocessed data for the current pipeline version
+        select * from sequence_entries_preprocessed_data
+        where pipeline_version = (select version from current_processing_pipeline
+                                  where organism = (
+                                      select organism from sequence_entries se
+                                      where se.accession = sequence_entries_preprocessed_data.accession
+                                      and se.version = sequence_entries_preprocessed_data.version
+                                  ))
```

Comment:
Could we keep this simpler and use a less nested query? Would this be logically equivalent?

```suggestion
        select sepd.*
        from
            sequence_entries_preprocessed_data sepd
            join sequence_entries se on sepd.accession = se.accession and sepd.version = se.version
            join current_processing_pipeline cpp on se.organism = cpp.organism
                and sepd.pipeline_versioin = cpp.version
```

Reply:
https://github.com/loculus-project/loculus/pull/3534/commits/9c8e7cac8bc748c5a2c544339d0d522262f66d8c

much better